wgengine/magicsock/relaymanager.go

```diff
@@ -4,11 +4,19 @@
 package magicsock
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
 	"net/netip"
 	"sync"
+	"time"
 
 	"tailscale.com/disco"
+	"tailscale.com/net/udprelay"
```

Comment:
I doubt we want to import this on all platforms. We really only need `udprelay.ServerEndpoint` in the current state, which could move or be duplicated to a shim.

Reply:
:question: Out of curiousity, do you mean that we should shim out this code to save space in the final binary, say for embedded systems?

Since the UDP relay is a kind of reverse exit node, it would be nice to have it available on any client that has the capability to `--advertise-exit-node`.